### ISSUE TITLE
net-snmp: use --netsnmp-agent-libs

### ISSUE THREAD

**Issue by zeha** [PowerDNS/pdns#7762]
On Debian sid:

```
$ net-snmp-config --agent-libs 
-Wl,-z,relro -Wl,-z,now -L/usr/lib/x86_64-linux-gnu -lnetsnmpmibs -lsensors -lpci -ldl -lnetsnmpagent -lwrap -Wl,-E -lnetsnmp -lcrypto -lm
```

```
$ net-snmp-config --netsnmp-agent-libs
-L/usr/lib/x86_64-linux-gnu -lnetsnmpmibs -lnetsnmpagent -lnetsnmp
```

from man net-snmp-config:
```
       --netsnmp-agent-libs
              lists netsnmp specific agent libraries
```

There's also `--base-cflags` but that doesn't appear to be very useful at this time.

**Comment by Habbie** [comment]
macOS:
```
[peter:~] $ net-snmp-config --netsnmp-agent-libs
-lnetsnmpmibs -lnetsnmpagent -lnetsnmp
[peter:~] $ net-snmp-config --base-cflags
-DNETSNMP_ENABLE_IPV6 -fno-strict-aliasing -g -Os -pipe -DBUILD=153 -DMACOSX_DEPLOYMENT_TARGET=10.12 -F/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.Internal.sdk/System/Library/PrivateFrameworks/ -F/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.Internal.sdk/System/Library/Frameworks/ -I/usr/local/libressl/include -Udarwin16 -Ddarwin16=darwin16 -I/usr/local/include
[peter:~] $
```

so yes to (1) and no to (2) indeed

**Comment by rgacogne** [comment]
`--netsnmp-agent-libs` looks good on Arch too, while `--base-cflags` matches `--cflags` exactly.

**Comment by zeha** [comment]
It looks like we don't need the CFLAGS from net-snmp at all. `snmp-agent.hh` includes `net-snmp-config.h` which defines `NETSNMP_USE_INLINE` and `NETSNMP_ENABLE_IPV6` (on buster), and the other CFLAGS in `--cflags` appear to be useless to us.

**Comment by rgacogne** [comment]
Isn't `--cflags` suppose to have the required include paths, if any?

**Comment by zeha** [comment]
Right. :(

**Comment by rgacogne** [comment]
Implemented in #7818, closing.
